src/network/core/tcp_admin.cpp

```diff
@@ -47,7 +47,13 @@ NetworkRecvStatus NetworkAdminSocketHandler::HandlePacket(Packet *p)
 {
 	PacketAdminType type = (PacketAdminType)p->Recv_uint8();
```

Comment:
If the client has quit, should we even receive the packet type? Maybe place the HasClientQuit() check above here?

Reply:
Recv_uint8 itself can set HasClientQuit to true.
The result is 0 (i.e. not valid) whether HasClientQuit was set before or during the call.

Putting the check below is to avoid logging a "received invalid packet type 0" message, when no such packet was ever received.

Reply:
Ah ha